### PR TITLE
[438] feat: adopt dot notation styling

### LINKED ISSUE #438: Adopt CSS dot notation on style naming
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
[Add any other context or screenshots about the feature request here.](https://discord.com/channels/1031982848485359626/1210373157240963123)

### PR DESCRIPTION
**Pull request by justEhmadSaeed**
## Describe your changes
- This PR allows the dot notation CSS class-based styling while introducing the `className` attribute to the `styles`.
- It deprecates the `names` attribute of `styles`.
- It fixes some linting errors as well.

### Screenshots [Optional]
image

## Issue ticket number and link

Closes #438 

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have added tests
- [ ] I have added a changeset `pnpm changeset add`
- [x] I have added example usage in the kitchen sink app

### CODE REVIEW

File: packages/framework/src/hooks/useRegisterBindings.ts [hunk 1]
```diff
@@ -28,10 +28,18 @@ export const useRegisterBindings = <T extends Record<string, unknown>>(
...
+  let classStyles = get(values, ["styles", "class"]) as unknown;
+  if (isString(classStyles) && classStyles.length > 0) {
+    classStyles = classStyles
+      .trim()
+      .split(" ")
+      .map((className) => `.${className}`)
+      .join(" ");
+  }
+  const styleNames = `${isString(namedStyles) ? namedStyles : ""} ${isString(classStyles) ? classStyles : ""}`;
+  const styleProperties = useStyleNames(styleNames);
```
Comment (evshi) [inline packages/framework/src/hooks/useRegisterBindings.ts:42]: Let's move this logic into useStyleNames

**Review by evshi — COMMENTED** [review]
Please use `className` attributes instead per https://github.com/EnsembleUI/ensemble/issues/1295?notification_referrer_id=NT_kwDOAD8NL7I5NzU5OTU0NzYxOjQxMzIxNDM#issuecomment-1986100333

File: packages/framework/src/hooks/useRegisterBindings.ts (hunk 1 above)
Comment (justEhmadSaeed) [reply packages/framework/src/hooks/useRegisterBindings.ts]: @evshi done with both changes

File: packages/runtime/src/shared/types.ts [hunk 2]
```diff
@@ -6,8 +6,12 @@ import type {
 } from "@ensembleui/react-framework";
 import type { HasBorder } from "./hasSchema";
 
+/**
+ * @deprecated Use the `class` attribute for styling instead of `names`.
+ */
 export type EnsembleWidgetStyles = Omit<React.CSSProperties, "direction"> & {
   names?: Expression<string>;
+  class?: Expression<string>;
```
Comment (evshi) [inline packages/runtime/src/shared/types.ts:14]: Should be `className`

File: packages/runtime/src/shared/types.ts [hunk 3]
```diff
@@ -6,8 +6,12 @@ import type {
 } from "@ensembleui/react-framework";
 import type { HasBorder } from "./hasSchema";
 
+/**
+ * @deprecated Use the `class` attribute for styling instead of `names`.
+ */
 export type EnsembleWidgetStyles = Omit<React.CSSProperties, "direction"> & {
   names?: Expression<string>;
```
Comment (evshi) [inline packages/runtime/src/shared/types.ts:13]: Move `deprecated` comment block to just the attribute.

File: packages/runtime/src/shared/types.ts (hunk 2 above)
Comment (justEhmadSaeed) [reply packages/runtime/src/shared/types.ts]: my bad, fixed it.

**Review by evshi — APPROVED** [review]

**Pull request merged**